### PR TITLE
QP-787: Update the documentation for the assistant API

### PR DESCRIPTION
**Pull request by bzvestey**
Adds the new surface area for locking to the assistant API documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -386,6 +386,26 @@ The `Workspace` object:
     modelServiceId: string,
     logicServiceId: string,
 
+    //
+    // Locking information
+    //
+    async lockedBy() {
+      // Returns the e-mail of the user who locked the Workspace, or null if the
```
Comment (mechaffin) [inline README.md:393]: nit: e-mail address

File: README.md [hunk 2]
```diff
@@ -637,6 +677,26 @@ The `Function` object:
       id: string
       name: string
     }
+
+    //
+    // Locking information
+    //
+    async lockedBy() {
+      // Returns the e-mail of the user who locked the Function, or null if the
```
Comment (mechaffin) [inline README.md:685]: nit: e-mail address

File: README.md [hunk 3]
```diff
@@ -896,14 +956,60 @@ Moves a collection of Kinds and Functions from the origin Workspace to the targe
...
+
+#### addLockingChangedListener = cb =>
+
+Registers a callback function with the locking changed event. When the currently
+active Workspace or its Knowledge Graphs and Functions change the callback
+function will be called with the `LockingChanged` object. Returns undefined.
+
+```js
+const lockingChangedCB = ({locks}) => {
+  if(locks.workspace) console.log('WORKSPACES CHANGED',locks.workspace)
```
Comment (mechaffin) [inline README.md:989]: nit: odd spacing here, but it is consistent within the example

Comment (bzvestey) [reply README.md:989]: Yeah, this was pulled from one of the other examples with updated names.  Being as it was written in a MD file by someone who does not work on the UI team, it does not follow our usual pattern.  I will fix some of the spacing here.

File: README.md (hunk 1 above)
Comment (bzvestey) [reply README.md:393]: I will change it, but any specific reason why?  Doing it with just e-mail seems to be to common way, and at this point adding address actually make it seem a little weird to me.

**Pull request merged**